### PR TITLE
Internal: Support pseudo classes and breakpoints for local element classes [ED-25198]

### PR DESCRIPTION
**Pull request by eavichay-elementor**
## 1. Problem & Context

Support pseudo-classes (`:hover`, `:focus`, `:active`) and breakpoint-specific styles in local element classes via plain CSS strings instead of property→value maps. Enables inline responsive/stateful styling during composition, eliminating post-build tweaks.

## 2. What Changed (Where)

| File | Change |
|------|--------|
| `style-applier.php` | Rewrote `apply()` signature: accepts CSS strings + `style_apply_mode` ('patch'/'replace'); parses via `Style_Variants_Merger`, builds multi-breakpoint/multi-state variants |
| `build-composition-ability.php` | Pass active breakpoints to `Style_Applier`, removed `element_types` param; updated schema doc to reflect CSS string format |
| `manage-elements-ability.php` | Added `style_apply_mode` param to update op; swapped style param from object to string; injects breakpoints into applier |
| `style-variants-merger.php` | Extracted `parse_css_string()` static method; lazy-loads converter; delegates CSS split/parse to `Css_Media_Splitter` + per-block conversion |
| `manage-classes-ability.php` | Removed `Css_Media_Splitter` import (now in merger); calls `Style_Variants_Merger::parse_css_string()` directly |
| Tests (3 files) | Added 253 lines: unit tests for applier/merger; integration tests for build/manage with pseudo-states, breakpoints, replace mode |
| Docs | Updated `.md`: clarified CSS string format, `@media(--breakpoint)` syntax, warned about write-conflict post-build |

## 3. How It Works

**Entry**: API caller provides `style: "color: red; &:hover { color: blue; } @media(--mobile) { font-size: 1rem; }"`.

**Parse**: `Style_Variants_Merger::parse_css_string()` → `Css_Media_Splitter::split()` extracts `@media(--X)` blocks → per-breakpoint `Css_Converter::parse_nested()` extracts `&:state` blocks → yields `[breakpoint_blocks, removal_breakpoints]`.

**Merge**: `apply_mode('patch'|'replace')` reconciles existing variants against incoming; 'patch' unions, 'replace' discards affected breakpoints before merge.

**Write**: Upsert style entry with merged variants; auto-register style ID in element's `classes.value`.

**Edge cases**: Empty CSS in 'replace' mode clears all variants; invalid breakpoint names reject early (before converter instantiation, cheap fail).

## 4. Risks

- **Backwards compat**: Old API tests removed (`.md` migration path only). Callers using `style: { prop: value }` will fail silently if not revalidated.
- **Lazy converter**: Factory callable `$get_converter` isn't invoked if media split fails—subtle but safe for parse errors. Document intent.
- **Variant identity**: No dedup on pseudo-state+breakpoint combos; caller can submit `&:hover @media(--mobile)` nesting, relies on downstream schema validation.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: modules/mcp/static-resources/abilities/build-composition.md [hunk 1]
```diff
@@ -218,14 +225,8 @@ Section with heading + button (NO explicit heights - content sizes naturally):
...
-    "Main Section": {
-      "padding": "6rem 4rem",
-      "background": "linear-gradient(135deg, #faf8f5 0%, #f0ebe4 100%)"
-    },
-    "Section Title": {
-      "font-size": "3.5rem",
-      "color": "#2d2a26"
-    }
+    "Main Section": "padding: 6rem 4rem; background: linear-gradient(135deg, #faf8f5 0%, #f0ebe4 100%);",
+    "Section Title": "font-size: 3.5rem; color: #2d2a26;"
```
Comment (Omerisra6) [inline modules/mcp/static-resources/abilities/build-composition.md]:
```suggestion
    "Main Section": "padding: 6rem 4rem; background: linear-gradient(135deg, #faf8f5 0%, #f0ebe4 100%); @media(--mobile) { padding: 3rem 1.5rem; }",
    "Section Title": "font-size: 3.5rem; color: #2d2a26; &:hover { color: var(--wc26-gold); } @media(--mobile) { font-size: 2.25rem; } @media(--tablet) { font-size: 2.75rem; }"
```

**Review by Omerisra6 — APPROVED** [review]
